src/services/services.ts

```diff
@@ -2365,6 +2369,40 @@ module ts {
                 return result;
             }
 
+            function isCompletionListBuilder(previousToken: Node): boolean {
+                if (previousToken) {
+                    var containingNodeKind = previousToken.parent.kind;
+                    switch (previousToken.kind) {
+                        case SyntaxKind.CommaToken:
+                            return containingNodeKind === SyntaxKind.CallExpression

```

Comment:
each of these cases should have a comment explaining why we want a builder here.